src/Error/UnexpectedResponse.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types=1);
+
+namespace Spawnia\Sailor\Error;
+
+use Psr\Http\Message\ResponseInterface;
+
+class UnexpectedResponse extends \Exception
+{
+    public int $statusCode = 0;
+
+    public string $responseBody = '';
+
+    /** @var array<string, list<string>> */
+    public array $responseHeaders;
```

Comment:
Why only set a default for some? I propose we do not set any defaults and ensure initialization through a constructor.

Reply:
I wonder if it would be extensible? E.g. when `ClientExceptionInterface` we might not have those properties like responseBody so not sure if it's correct to require them via constructor.

Reply:
I would assume `UnexpectedResponse` exceptions only to occur when the server actually delivered a response. If the underlying HTTP failed due to other reasons (network issues, DNS, missing/malformed URL), i would expect a different exception class altogether.

Reply:
Reasonable. I'll use constructor